erpnext/agriculture/doctype/plant_batch/plant_batch.py

```diff
@@ -15,6 +15,14 @@
 
 class PlantBatch(Document):
 	def validate(self):
+		if self.project:
+			tasks = frappe.get_all("Task", {"project":self.project})
+			for task in tasks:
+				if self.start_date:
+					frappe.db.set_value("Task", task.name, "exp_start_date", self.start_date, update_modified=False)
```

Comment:
Don't do a `db.set_value`, it's better to get the task doc, update and then call `save`. That'll also do any controller validations.

And the `update_modified=False` should only be set for patches, not actual logic, since we do want to update the last modified here.